spec/models/user_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+
+  describe 'DB table' do
+    it { is_expected.to have_db_column :name }
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :email }
+  end
+
+  describe 'mailboxer' do
+    let(:sender) {build(:user, name: 'sender') }
+    let(:receiver) {build(:user, name: 'receiver')}
+
+    before do
+      sender.send_message(receiver, "body", "subject")
+    end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+  end
+
+    it 'creates a user with valid info' do
+      expect(sender).to be_valid
+    end
+
+    describe 'send message' do
+
+      it 'should be a new message in the Recievers inbox' do
+        expect(receiver.mailbox.inbox.count).to eq 1
+      end
+
+      it 'should be a new message in the Senders outbox' do
+
+        expect(sender.mailbox.sentbox.count).to eq 1
+      end
+
+      it 'should have a subject' do
+        expect(receiver.mailbox.inbox.last.subject).to eq "subject"
+      end
+
+      it 'should have a text body' do
+        conversation = receiver.mailbox.inbox.first
+        receipts = conversation.receipts_for receiver
+        receipts.each do |receipt|
+          @message = receipt.message
+        end
```

Comment:
Is this really needed? Why do you iterate over this instead of saying `@message = receipts.first.message`? Also, why do you need to store it in an instance variable?

Reply:
I don't really know. This was the way Raoul showed us yesterday, quite frankly I don't understand it fully.